api/types/types.go

```diff
@@ -461,13 +461,23 @@ type Runtime struct {
 	Args []string `json:"runtimeArgs,omitempty"`
 }
 
+// DownloadCacheUsage contains information about the disk usage of the download
+// cache
+type DownloadCacheUsage struct {
+	TotalNumber  int64
+	TotalBytes   int64
+	ActiveNumber int64
+	ActiveBytes  int64
+}
+
 // DiskUsage contains response of Engine API:
 // GET "/system/df"
 type DiskUsage struct {
-	LayersSize int64
-	Images     []*ImageSummary
-	Containers []*Container
-	Volumes    []*Volume
+	LayersSize         int64
+	Images             []*ImageSummary
+	DownloadCacheUsage DownloadCacheUsage
```

Comment:
Yeah, there's no way to associate this data with any particular image and even if there was, the image wouldn't exist in that array. I don't expect us to have other types of caches right now. This feels like premature generalization. If you can give me a good example of another type of cache we might add, I can make it a map or array.